app/ante/fee_checker.go

```diff
@@ -29,7 +30,7 @@ func CheckTxFeeWithGlobalMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, in
 	// global minimum fee only applies to app versions greater than one
 	if ctx.BlockHeader().Version.App > v1.Version {
 		// convert the global minimum gas price to a big.Int
-		globalMinGasPrice, err := sdk.NewDecFromStr(fmt.Sprintf("%f", appconsts.GlobalMinGasPrice))
+		globalMinGasPrice, err := sdk.NewDecFromStr(fmt.Sprintf("%f", v2.GlobalMinGasPrice))
```

Comment:
[not blocking][question] if the constant is now app version specific, should we define a getter like this in `versioned_consts.go`:

```go
func GlobalMinGasPrice(version uint64) (float64, error) {
	switch version {
	case v2.Version:
		return v2.GlobalMinGasPrice, nil
	default:
		return 0, fmt.Errorf("global min gas price not defined for version %d", version)
	}
}
```

and then use here

```go
	// global minimum fee only applies to app versions greater than one
	if ctx.BlockHeader().Version.App > v1.Version {
		gmgp, err := appconsts.GlobalMinGasPrice(ctx.BlockHeader().Version.App)
		if err != nil {
			return nil, 0, errors.Wrap(err, "invalid GlobalMinGasPrice")
		}

		// convert the global minimum gas price to a big.Int
		globalMinGasPrice, err := sdk.NewDecFromStr(fmt.Sprintf("%f", gmgp))
		if err != nil {
			return nil, 0, errors.Wrap(err, "invalid GlobalMinGasPrice")
		}
```

Reply:
Feel free to rename variables, add godocs, etc. Mostly for illustrative purposes ^

Reply:
thanks for the review and suggestions! I completely missed the `versioned_consts.go` file 🤦🏻‍♀️ updated now

Reply:
No problemo. I wonder if the appconsts in v1 and v2 should actually be un-exported so that consumers are forced to use the getters rather than hard-coding a particular versioned constant.

I know Callum found a few instances where code was relying on a default or a particular versioned constant rather than querying for a versioned constant based on app version. Can discuss more outside this PR if maintainers think we should pursue this idea.

Reply:
yes, I think making them unexported makes sense. we should probably go through the app and replace all hardcoded values with getters+constants based on app version as you suggested. cc @cmwaters @evan-forbes